master/src/main/java/org/apache/celeborn/service/deploy/master/clustermeta/ha/StateMachine.java

```diff
@@ -216,16 +222,17 @@ private synchronized void loadSnapshot(SingleFileSnapshotInfo snapshot) throws I
     try {
       setLastAppliedTermIndex(snapshot.getTermIndex());
       install(snapshotFile);
-    } catch (Exception e) {
-      throw new IOException(String.format("Failed to load snapshot %s", snapshot), e);
+    } catch (IOException e) {
+      throw new IOException(String.format("Failed to load snapshot %s", snapshot), e.getCause());
     }
   }
 
-  private void install(File snapshotFile) {
+  private void install(File snapshotFile) throws IOException {
     try {
       metaHandler.loadSnapShot(snapshotFile);
-    } catch (Exception e) {
+    } catch (IOException e) {
```

Comment:
I suggest to name a exception "rethrow" if we just catch and re-throw it.

Reply:
Done